src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.carnivore = true;
+  }
+
+  bite(animalName) {
+    if (!animalName.hidden && !animalName.carnivore) {
```

Comment:
```suggestion
    if (!animalName.hidden && !animalName.carnivore) {
```
Here you can check if the animal is instaceOf Herbivore.